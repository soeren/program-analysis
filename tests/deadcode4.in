// Check for dead code elimination
// - Check for correct inter-procedural elinimination
// - Check that dead branches are removed

program

int x;
int y;
int z;

int stub(int a){
    int f; // dead
    f := a*a; // dead
    if(false){
        return a*a; // dead branch -> only ELSE branch should stay
    } else {
        return a;
    }
}

x := stub(123);
y := 666 * x;

if(true){
    z := 42;
    x := 11;
} else { // dead branch
    z := 0;
}

x := 77*z;

while(false){ // dead branch
    z := stub(x);
}

while(x > y){ // stays unchanged
    x := x - 2;
    x := x - 1;
}

write(x);

end
