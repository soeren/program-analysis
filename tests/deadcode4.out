program
int x;
int y;
int z;
int stub(int a) // 9
{
     return a; // 10
 }
x := stub(123); // 1
y := 666 * x; // 2
z := 42; // 3
x := 77 * z; // 4
while(x > y) // 5
{
     x := x - 2; // 6
     x := x - 1; // 7
 }
write(x); // 8
end
